rootex/framework/components/visual/model/sprite_component.cpp

```diff
@@ -0,0 +1,127 @@
+#include "sprite_component.h"
+
+#include "common/common.h"
+
+#include "event_manager.h"
+#include "core/resource_loader.h"
+#include "systems/render_system.h"
+
+DEFINE_COMPONENT(SpriteComponent);
+
+SpriteComponent::SpriteComponent(Entity& owner, const JSON::json& data)
+    : RenderableComponent(owner, data)
+    , m_ImageMaterialResourceFile(ResourceLoader::CreateBasicMaterialResourceFile("rootex/assets/materials/sprite.basic.rmat"))
+{
+	configureBoundingRectangle();
+}
+
+void SpriteComponent::configureBoundingRectangle()
+{
+	float m_ImageScaleFactor = 0.001;
+
+	Ref<Texture> imageTexture = m_ImageMaterialResourceFile->getDiffuseImageFile()->getTexture();
+	float rectWidth = imageTexture->getWidth() * m_ImageScaleFactor;
+	float rectHeight = imageTexture->getHeight() * m_ImageScaleFactor;
+
+	PRINT("Rect Width: " + std::to_string(rectWidth));
+	PRINT("Rect Height: " + std::to_string(rectHeight));
+
+	Vector<VertexData> vertices;
+	VertexData vertex;
+	ZeroMemory(&vertex, sizeof(VertexData));
+	for (int i = 0; i < 4; i++)
+	{
+		int u = (i == 1 || i == 2) ? 1 : 0;
+		int v = (i == 2 || i == 3) ? 1 : 0; 
+
+		vertex.position.x = u * rectWidth / 2;
+		vertex.position.y = v * rectHeight / 2;
```

Comment:
why `/ 2`?

Reply:
Was a mistake, corrected it.
Also should I center it about the entity's transform?